packages/components/src/public/QueryModel/withQueryModels.tsx

```diff
@@ -755,13 +847,29 @@ export function withQueryModels<Props>(
                         // We need to reset all data for the model because changing the view will change things such as
                         // columns and rowCount. If we don't do this we'll render a grid with empty rows/columns.
                         resetRowsState(model);
+                        resetTotalCountState(model);
                         resetSelectionState(model);
                     }
                 }),
                 () => this.maybeLoad(id, false, shouldLoad, shouldLoad && loadSelections)
             );
         };
 
+        /**
+         * Reset the totalCount state for all models so that the next time loadModel or loadAllModels() is called,
+         * it will also call the loadTotalCount().
+         */
+        resetTotalCountState = (): void => {
+            Object.keys(this.state.queryModels).forEach(id => {
```

Comment:
I would recommend consolidating this into a single state update as opposed to a call to setState per model.

Reply:
done